docs/components/configurationPanel.md

```diff
@@ -0,0 +1,224 @@
+    description: {
+        buttons: { "16": "Button 16: Badge Icon Button" },
+        axes: { "0": "Left Thumbstick Horizontal Direction" }
+    },
+    message: {
+        buttons: {
+            null: "None",
+            scrollLeft: "Scroll left",
+            scrollRight: "Scroll right"
+        },
+        axes: {
+            null: "None",
+            thumbstickTabbing: "Focus on the previous/next element"
+        }
+    },
+    actions: {
+        speedFactorOption: ["scrollLeft", "scrollRight", "thumbstickTabbing"],
+        invertOption: ["thumbstickTabbing"]
+    }
+});
+```
+
+## Invokers
+
+### `{configurationPanel}.createMenu()`
+
+- Returns: Nothing.
+
+Creates a separate configuration menu for each gamepad input, as mentioned in the `description` option and is triggered
+when the component is created. This invoker calls other invokers to create the configuration options and to listen to
+various interaction-based events and changes on the configuration panel.
+
+### `{configurationPanel}.createInputActionDropdown(inputIdentifier, configMenu, inputType, currentValue)`
+
+- `inputIdentifier {String}` A description of the gamepad input type and its index. For example, "button-0".
+- `configMenu {Object}` The DOM Object used for storing the configuration options for a given gamepad input.
+- `inputType {Boolean}` The type of the gamepad input, i.e. "buttons" or "axes".
+- `currentValue {Object}` The value of the currently selected action for the given gamepad input.
+- Returns: Nothing.
+
+Creates an "Action" label and dropdown for the given gamepad input and then injects these elements into the `configMenu`
+object passed as an argument. The attributes and classnames of these elements are also configured before they are
+injected, and the value of the "Action" dropdown is set equal to the currently selected action.
+
+### `{configurationPanel}.createSpeedFactorOption(inputIdentifier, configMenu, currentValue)`
+
+- `inputIdentifier {String}` A description of the gamepad input type and its index. For example, "button-0".
+- `configMenu {Object}` The DOM Object used for storing the configuration options for a given gamepad input.
+- `currentValue {Object}` The current "Speed Factor" value for the given gamepad input.
+- Returns: Nothing.
+
+Creates a "Speed Factor" label and input field for the given gamepad input and then injects these elements into the
+`configMenu` object passed as an argument. The attributes and classnames of these elements are also configured before
+they are injected and the value of the "Speed Factor" input field is set equal to the current `speedFactor` value.
+
+### `{configurationPanel}.createCheckbox(inputIdentifier, configMenu, isAxes, currentValue)`
+
+- `inputIdentifier {String}` A description of the gamepad input type and its index. For example, "button-0".
+- `configMenu {Object}` The DOM Object used for storing the configuration options for a given gamepad input.
```

Comment:
Same comment here about documenting the fact that you modify the argument passed to you.